client/src/components/graph/graph.tsx

```diff
@@ -929,24 +963,82 @@ class Graph extends React.Component<GraphProps, GraphState> {
     return createColorQuery(colorMode, colorAccessor, schema, genesets);
   }
 
+  updateOpenSeadragon() {
+    const {
+      viewport: { width, height },
+    } = this.state;
+
+    const {
+      config: { s3URI },
+    } = this.props;
+
+    if (
+      this.openseadragon ||
+      !shouldShowOpenseadragon(this.props, this.state) ||
+      !width ||
+      !height
+    ) {
+      return;
+    }
+
+    this.openseadragon = Openseadragon({
+      // (thuang): This id will need to be unique for multiple graphs
+      id: "openseadragon",
+      prefixUrl: getSpatialPrefixUrl(s3URI),
+      tileSources: getSpatialTileSources(s3URI),
+      showNavigationControl: false,
+    });
+
+    /**
+     * (thuang): Remove the openseadragon element when the image fails to load,
+     * likely because the image is not found in the S3 bucket.
+     */
+    this.openseadragon.addHandler("open-failed", () => {
+      this.setState((state) => ({
+        ...state,
+        isDeepZoomSourceValid: false,
+      }));
+
+      postUserErrorToast("The image is not available");
```

Comment:
when it fails to load, it will show a dismissable yellow banner

<img width="1233" alt="Screenshot 2024-05-10 at 2 17 53 PM" src="https://github.com/chanzuckerberg/single-cell-explorer/assets/6309723/41f27355-897c-4aaf-847e-82dfb3d148e9">